### PR TITLE
Fix Table.Row prop typings, part 1

### PR DESCRIPTION
**Pull request by vojtechszocs**
Originally, Console `Table.Row` prop used to reference React components.

Currently, using `VirtualTableBody.rowRenderer` from `@patternfly/react-virtualized-extension` implies that rows are callbacks, not React components.

The force cast

```ts
const row = (Row as RowFunction)(rowArgs as RowFunctionArgs);
```

is a work-around, along with

```ts
Row?: RowFunction | React.ComponentClass<any, any> | React.ComponentType<any>;
```

This PR fixes all row typings, minus a few convoluted cases which will be addressed in a follow-up.

cc @benjaminapetersen @spadgett @christianvogt @rawagner

### CODE REVIEW

**Comment by vojtechszocs** [comment]
/hold

After talking with @christianvogt we should reduce the changes, fixing our typings and components that cause the `key` prop access error.

This PR needs an update.

**Comment by vojtechszocs** [comment]
/hold cancel

PR updated.

File: frontend/public/components/factory/table.tsx [hunk 1]
```diff
@@ -328,16 +328,17 @@ const VirtualBody: React.SFC<VirtualBodyProps> = (props) => {
...
   columns: [];
   isScrolling: boolean;
   key: string;
   style: object;
-  customData?: any;
+  customData?: C;
 };
-export type RowFunction = (args: RowFunctionArgs) => JSX.Element;
+
+export type RowFunction<T = any, C = any> = (args: RowFunctionArgs<T, C>) => JSX.Element;
```
Comment (vojtechszocs) [inline frontend/public/components/factory/table.tsx]: @christianvogt Should we change this to `=> React.ComponentType<any>` ?

Comment (christianvogt) [reply frontend/public/components/factory/table.tsx]: It should return `React.ReactNode`

Comment (vojtechszocs) [reply frontend/public/components/factory/table.tsx]: Agreed, let's change this as part of follow-up PR (part 2) which will address the convoluted cases.

**Pull request merged**